ClientLib/src/main/java/com/o3dr/android/client/utils/data/tlog/TLogParser.java

```diff
@@ -0,0 +1,349 @@
+            iteratorExecutor.shutdownNow();
+            in.close();
+        }
+
+        /**
+         * Retrieve next message from TLog file asynchronously.
+         * start must be called before this method.
+         *
+         * @param callback
+         */
+        public void nextAsync(final TLogIteratorCallback callback) {
+            iteratorExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        sendResult(callback, next(in));
+                    } catch (EOFException e) {
+                        sendResult(callback, null);
+                    } catch (IOException e) {
+                        sendFailed(callback, e);
+                    }
+                }
+            });
+        }
+
+        /**
+         * Retrieve next message with specified message filter from TLog file asynchronously.
+         * start must be called before this method.
+         *
+         * @param filter
+         * @param callback
+         */
+        public void nextAsync(final TLogIteratorFilter filter, final TLogIteratorCallback callback) {
+            iteratorExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        Event event = null;
+                        while (in.available() > 0) {
+                            Event currEvent = next(in);
+                            if (filter.acceptEvent(currEvent)) {
+                                event = currEvent;
+                                break;
+                            }
+                        }
+                        sendResult(callback, event);
+                    } catch (EOFException e) {
+                        sendResult(callback, null);
+                    } catch (IOException e) {
+                        sendFailed(callback, e);
+                    }
+                }
+            });
+        }
+
+        private void sendResult(final TLogIteratorCallback callback, final Event event) {
+            if (callback != null) {
+                handler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        callback.onResult(event);
+                    }
+                });
+            }
+        }
+
+        private void sendFailed(final TLogIteratorCallback callback, final Exception e) {
+            if (callback != null) {
+                handler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        callback.onFailed(e);
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Returns a list of all events in specified TLog uri
+     *
+     * @param uri
+     * @param callback
+     */
+    public static void getAllEventsAsync(final URI uri, final TLogParserCallback callback) {
```

Comment:
@chaviw this method should require a `Handler` parameter that it uses to dispatch the result back to the handler thread.